### ISSUE TITLE
Module system: Rename `@forward show/hide` to `@export only/except`

### ISSUE THREAD

**Issue by glebm** [sass/language#46]
For the `@forward "some-module" show/hide thing` syntax, consider using `only` / `except` instead of `show` / `hide`. Also consider `@export` instead of `@forward`.

`only` and `except` are immediately clear even to those who do not fully understand the module system and haven't used languages with such terminology; `only` also indicates that no other names get exported except the listed ones.

It also reads better, VERB NOUN ADVERB NOUN instead of the somewhat clunky VERB NOUN VERB NOUN.

`@export` is similar to JavaScript, which most Sass users are likely to be at least a bit familiar with.

**Comment by ArmorDarks** [comment]
To be honest, the whole idea with the current way module system works (with `forward`) is kinda extravagant, and it might turn to be way too extravagant.

I'm not skilled enough in language design to voice my concerns in a more constructive way, but it feels like forward implies `all included by default` model, which is _not explicit_ on contrary to traditional imports and exports. And inexplicity is known to bring a lot of troubles, alongside with unobvious magic. Is it too late to review it?

I fell also that attempt to make module system work whole differently from how traditional modules works is even worse in terms that it makes things inconsistent. Sass modules will work in `their own way`. Why not try to apply already existing and commonly accepted modules standards?

**Comment by nex3** [comment]
@ArmorDarks It is definitely too late to change such a major aspect of the proposal. The time to bring up issues like that was when the RFC period.

I'm not sure exactly what you mean by an "traditional imports and exports", but I'm guessing you're referring to a JS-style module system where nmaes are hidden by default unless they're explicitly export. We didn't go with that approach for reasons outlined in [the first Frequently Asked Question](https://github.com/sass/language/blob/master/proposal/module-system.md#frequently-asked-questions).

I also think you're wrong that JS's model is "traditional" in any sense. JS is practically unique among languages in requiring exports to be marked explicitly, and it came about more as a historical legacy of building a module system as a library (require.js) without being able to modify the language in any way. Essentially all other widely-used languages either make module members public by default or require every member to be explicitly declared "public" or "private".

**Comment by ArmorDarks** [comment]
I agree that definitely too late. Didn't have time to write about sooner, unfortunately. But I even didn't try, to be honest, because when I was reading fist spec it was already clear that decision has been made. It was obvious, that something like JS modules system already been taken into account and thrown away.

> We didn't go with that approach for reasons outlined in the first Frequently Asked Question.

I've read the spec and FAQ, of course. The argumentation didn't cut that for me, though. That's why the question about whether it was indeed good choice remained in my head.

> I'm not sure exactly what you mean by an "traditional imports and exports", but I'm guessing you're referring to a JS-style module system

Yeap, I'm mostly referring to the JS modules spec. As a related to the web technology, it's somehow expected to have a similar spec in other related techs, such as Sass.

> I also think you're wrong that JS's model is "traditional" in any sense.

I'm probably starting senseless offtopic, but I can't agree with that statement. Most language with import mechanism function in very similar fashion in the sense that they all define explicitly what is public and what is not. Even languages, which don't have straight `export` or something like that operator, usually operate on other levels — as through private and public operators, etc, but still remain true to the explicity of the declarations of what usable outside and what is not. So, they do not export all by default, as it proposes Sass spec.

It is just that happens that JS received some more modern and more obvious modules system, which, in my opinion, expressing dependencies better and in a more effective way, unless you're not in traditional OOP-land...
